### PR TITLE
Clients: Fix installation issues with Py2->Py3 and hash sum #2480

### PR DESCRIPTION
**Pull request by XeBoris**
There are two issues observed in: 
  - Installation in Python 3.6.8 failed because of bad translated Py2->Py3 code in rucio/lib/rucio/rse/protocols/storm.py when it comes to evaluate the exceptions.
  - The hash sum calculation failed due to an encoding error when the hash sum is calculated for the final path definition. (protocols.py)

Both are addressed in this pull request.

### CODE REVIEW

File: lib/rucio/client/uploadclient.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 
 class UploadClient:
 
-    def __init__(self, _client=None, logger=None, tracing=True):
+    def __init__(self, _client: object = None, logger: object = None, tracing: object = True) -> object:
```
Comment (bari12) [inline lib/rucio/client/uploadclient.py]: The type hints are not backwards compatible unfortunately 😞

File: lib/rucio/client/uploadclient.py [hunk 2]
```diff
@@ -312,8 +315,11 @@ def _register_file(self, file, registered_dataset_dids):
             meta = self.client.get_metadata(file_scope, file_name)
             logger.info('File DID already exists')
             logger.debug('local checksum: %s, remote checksum: %s' % (file['adler32'], meta['adler32']))
+
             if meta['adler32'] != file['adler32']:
+                logger.info("Here")
```
Comment (bari12) [inline lib/rucio/client/uploadclient.py]: Please remove

File: lib/rucio/common/pcache.py [hunk 3]
```diff
@@ -873,11 +873,20 @@ def daemonize(self):
...
+
+        #try to be backwards compatible with Python2 (?)
+        try:
+            for fd in xrange(0, maxfd + 1):
+                try:
+                    os.close(fd)
+                except:
+                    pass
+        except NameError:
+            for fd in range(0, maxfd + 1):
```
Comment (bari12) [inline lib/rucio/common/pcache.py]: Why not just use range in both cases?

**Review by bari12 — CHANGES_REQUESTED** [review]

File: lib/rucio/common/pcache.py (hunk 3 above)
Comment (XeBoris) [reply lib/rucio/common/pcache.py]: Indeed, this is much simpler.

File: lib/rucio/client/uploadclient.py (hunk 2 above)
Comment (XeBoris) [reply lib/rucio/client/uploadclient.py]: oh! I forgot to delete it before uploading.

File: lib/rucio/client/uploadclient.py (hunk 1 above)
Comment (XeBoris) [reply lib/rucio/client/uploadclient.py]: Right! I changed it back

**Comment by XeBoris** [comment]
I submitted the requested changes but the build still fails. The log files saying it is failing only for one test (py2.7 with Oracle) and Travis ends with these lines in the log file (everything before looks reasonable to me):
```
RSE (CLIENTS): get protocols for operations of rse. ... ok
RSE (CLIENTS): get protocols for operations of rse (RSEOperationNotSupported). ... 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

Furthermore, Python3.6 tests seems to fail since flake8 is missing:
```
===============================
Running flake8                 
===============================
./tools/travis/check_syntax.sh: line 23: flake8: command not found
```

**Comment by bari12** [comment]
Hi Boris,
Yes, the oracle test hangs sometimes, I am usually just restarting them, but now on vacation so I didn't see this right away.
The missing flake8 for py36 we know about. This will be fixed as well.
Thanks!

**Pull request merged**